daft/catalog/__init__.py

```diff
@@ -203,6 +203,9 @@ def register_python_catalog(catalog: object, name: str | None = None) -> str:
     return name
 
 
+Properties = dict[str, Any]
```

Comment:
nit: I tend to avoid type aliases that don't add semantic meaning (like `Properties = dict[str, Any]`) as they can add unnecessary indirection, but this is purely personal preference.

Reply:
Makes sense, this is pyiceberg style. Ideally I want these to be TypedDict at some point.